### DISCUSSION TITLE
How to calculate summary of percent field  of pivot relation ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#6957] [Help] [answered]
On Laravel/filament site I have a Ticket model with users pivot relation with UsersRelationManager:

```
    public static function getRelations(): array
    {
        return [
            UsersRelationManager::class,
        ];
    }

```
as  pivot has 'percent' I need calculate summary of percent field equals 100 or not and show it in header or footer(preferable) of
UsersRelation table below of the form. So I need :
1) Get some action to fill  header or footer of UsersRelation table
2) In this action I need to run through all UsersRelations row to calculate percent summary and formate output string to return

How  can I do it ?

```
        "laravel/framework": "^9.52.10",
        "filament/filament": "^2.17.49",
        "filament/forms": "^2.17.49",

```
Thanks in advance!

**Comment by pboivin** [comment, marked as answer]
You can dig into the RelationManager class to find all available options, but maybe this can give you a quick idea (add it to your `UsersRelationManager`):

```php
protected function getTableDescription(): string | Htmlable | null
{
    $count = $this->getTableQuery()->count();

    return "Total: $count";
}
```

There's also `getTableHeader()`, but I'm not sure about the footer area.

**Comment by PetroGromovo** [comment]
Thank you!
getTableDescription and getTableHeader are called.
But are they described in docs? I did not find...

**Reply by pboivin** [reply]
I'm not sure it's all documented, I found it by looking around in the code :)
